### ISSUE TITLE
Should my SERVE ARRIVE, etc. classes's access be public

### ISSUE THREAD

**Issue by dorothyyuann** [nus-cs2030/2122-s2#565]
Code Crunch states that only SimulateN should be public but if I remove public access from my constructors or main class I seem to have errors with running the program? Not sure what this means
<img width="1194" alt="Screenshot 2022-04-24 at 10 24 59 PM" src="https://user-images.githubusercontent.com/101258725/164981240-13646e5c-ede5-4b6d-908c-00094e2e29c2.png">

**Comment by denshem10** [comment]
What public-facing classes means is that you don't have to package it into either .simulator or .util.  
That's how I'm doing it, I haven't run into any problems yet.
So I think if you want your methods or the class itself to be public (mine is), it should be ok!

**Comment by FeanTMX** [comment]
For me I passed codecrunch with most of my classes public (i needed them to be public to do some tests in jshell). It didnt give me much problems so should be fine keeping them as public

**Comment by blmj00** [comment]
yup same here i made most of my classes and methods public with no issues

**Comment by joannelxl** [comment]
same here, making it public allows u to test out out your code in jshell too!

**Comment by jodytng** [comment]
hi! I made all my classes private except the SimulateN ones but I think it should be alright for you to leave them as public!

**Comment by fionaxjy** [comment]
Hi, I also made all my classes private. But I think it might be easier for you to leave them as public as there are fewer restrictions.

**Comment by AaronTay21** [comment]
hi i made my methods public

**Comment by Brianliew77** [comment]
yes u can make them public

**Comment by Alexius0123** [comment]
Hello! If you order your files in the correct way, you will find that the event classes are actually supposed to be private, and not have a public accessor. If they are still public by Level 9, you will be thrown an error that says something along the lines of missing Javadoc comments. The best way to order is to have one cs2030 main folder with all your .java files, and a cs2030 subfolder with 2 more subfolders of util and simulator to store your .class files.

Hope this helps!

**Comment by Zarilex09** [comment]
Yes its possible to make them public

**Comment by jonyeokj** [comment]
To add on to what everybody mentioned before, you have to make some of your classes and methods public otherwise they can't access one another. Personally, I had trouble running my code at first because I didn't make my methods public. I am also aware of the private level 9 test case that Alexius mentioned earlier of the error regarding the missing Javadoc files, so it's important to make sure to order your files properly like he suggested before.

**Comment by jiawei88** [comment]
yup you should make it all public due to the fact that are so many different classes and some of them could be packaged in a different folder. So for safety you public all your classes, methods, except of course your private final attributes :)

**Comment by jlee1007** [comment]
i have made my classes & constructors public & it passed CC!

**Comment by aldrichwilliams23** [comment]
Yes, I think it's fine to make your classes & constructors public, there should be no issue with the CodeCrunch test cases!
